lib/companies/companies.ex

```diff
@@ -126,9 +126,12 @@ defmodule Companies.Companies do
     Company.changeset(company, %{})
   end
 
-  defp predicates(query, %{"type" => <<start, finish>>}) when start in 97..122 do
+  defp predicates(query, %{"type" => <<start, finish>>}) do
     query = distinct(query, [c, _i, _j], asc: fragment("LOWER(?)", c.name))
 
+    start  = if (start  ==  97) do  32 else  97 end
```

Comment:
@RichMorin can you explain what you're trying to accomplish here? I think there may be a better way but it's not entirely clear the goal.

Reply:
The `start` and `finish` munging are supposed to extend the range of items displayed. The lowest range (currently `ae`) should display companies whose names begin with any characters that sort before `a` (e.g., `0`). The highest range (currently `uz`) should display companies whose names begin with any characters that sort after `z` (e.g., `{`).

Reply:
If we need to handle numbers and special characters, we should add that filter instead of putting numbers into A-E and symbols into U-Z.

Reply:
There are companies in the data set whose names begin with digits (e.g., "500 Startups") and special characters (e.g., "{sauce}"), so I don't think there is much question about whether they need to be displayed. Where to put them is a judgement call. If you're listing the entire data set, do you put them all at the end, split them up according to the ASCII sort sequence, or what?  And, if you're displaying range-based subsets such as "A-E", do you define a range for the digits and special characters.  Your call...